### PR TITLE
Fix memory_range, add some tests for it

### PR DESCRIPTION
**Pull request by 5225225**
There might be room for more tests here, but now at least the range is tested *somewhere*.

Fixes: #407

### CODE REVIEW

**Comment by 5225225** [comment]
This came up *again* in #413. I'll write an extension trait / helper function to do the validation of `Range::new` properly so it can never panic, and returns a `Option<Range>`, which is really what we want in all of these cases

**Review by luser — APPROVED** [review]
Thanks for adding extensive tests!

File: minidump/src/minidump.rs [hunk 1]
```diff
@@ -4994,6 +4994,20 @@ c70206ca83eb2852-de0206ca83eb2852  -w-s  10bac9000 fd:05 1196511 /usr/lib64/libt
...
         assert!(maps[1].is_shared);
         assert!(!maps[1].is_executable());
+
+        let mut unified_infos = unified_info.by_addr();
+
+        if let UnifiedMemoryInfo::Map(m) = unified_infos.next().unwrap() {
+            assert_eq!(m, maps[0]);
+        } else {
+            panic!("expected a map");
+        }
```
[the remark this answers is not in the document]
Comment (5225225) [reply minidump/src/minidump.rs:5004]: It compiles fine if I make `UnifiedMemoryInfo` implement `PartialEq` (Which is user-visible). You don't need PartialEq to match on enum variants like this.

Arguably, it maybe *should* anyways. But again, user visible change. Your call if I should leave it as-is or derive PartialEq (and Eq?)

**Comment by 5225225** [comment]
I'll make that extension trait when I fix #413, not here.

**Comment by luser** [comment]
> I'll make that extension trait when I fix #413, not here.

I'm not overly fussed about any of this, FWIW, since it's in tests.

File: minidump/src/minidump.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (5225225) [reply minidump/src/minidump.rs]: Ah, yeah, I keep on forgetting `matches!` exists. Changed.

**Comment by Gankra** [comment]
Awesome, thank you so much!!

**Pull request merged**